AmplifyPlugins/DataStore/AWSDataStoreCategoryPlugin/Sync/SubscriptionSync/AWSIncomingEventReconciliationQueue.swift

```diff
@@ -105,6 +108,11 @@ final class AWSIncomingEventReconciliationQueue: IncomingEventReconciliationQueu
 
     private func onReceiveValue(receiveValue: ModelReconciliationQueueEvent) {
         switch receiveValue {
+        case .finished:
+            _ = receivedCount.increment()
+            if receivedCount.get() == reconciliationQueues.count {
```

Comment:
This does not solve the race condition you're presumably trying to catch by using an AtomicValue. Further, relying just on counts feels like a brittle way of tracking completion. Let's discuss this design and see if there's some other way we can track this.